lib/apiHelpers.js

```diff
@@ -0,0 +1,26 @@
+// utils/apiHelpers.js
+import NextCors from "nextjs-cors";
+
+// Utility function to prepare global API settings
+async function makePublicEndpoint(req, res, allowedMethods = ["GET"]) {
+	const allowedMethodsWithOptions = ["OPTIONS", ...allowedMethods];
+
+	// Initialize CORS with common settings
+	await NextCors(req, res, {
+		methods: allowedMethodsWithOptions, // Options shall always be allowed
+		origin: "*", // Adjust according to your needs
+		optionsSuccessStatus: 200,
+	});
+
+	// Check for allowed HTTP methods
+	if (!allowedMethodsWithOptions.includes(req.method)) {
```

Comment:
I would split this into another helper function utility. I have it on a to-do somewhere to extract that into a separate function because we also use it on the internal API.

Reply:
Thx, split it out